src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIQueueWebsocketService.java

```diff
@@ -0,0 +1,132 @@
+package de.tum.in.www1.artemis.service.connectors.localci;
+
+import java.util.Objects;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import com.hazelcast.collection.IQueue;
+import com.hazelcast.collection.ItemEvent;
+import com.hazelcast.collection.ItemListener;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.map.listener.EntryAddedListener;
+import com.hazelcast.map.listener.EntryRemovedListener;
+
+import de.tum.in.www1.artemis.service.connectors.localci.dto.LocalCIBuildAgentInformation;
+import de.tum.in.www1.artemis.service.connectors.localci.dto.LocalCIBuildJobQueueItem;
+import de.tum.in.www1.artemis.web.websocket.localci.LocalCIBuildQueueWebsocketService;
+
+/**
+ * This service is responsible for sending build job queue information over websockets.
+ * It listens to changes in the build job queue and sends the updated information to the client.
+ * NOTE: This service is only active if the profile "localci" and "scheduling" are active.
```

Comment:
please explain why this is the case, mainly to avoid sending the exact same information multiple times